### PR TITLE
fix(discord): show actual tool name for single-tool-call turns

### LINKED ISSUE #886: feat: show actual tool name in Discord when only one tool call in a turn
## Summary

When a Foreman turn contains only a single tool call, Discord currently displays a generic label like **"bash x 1"** instead of the actual tool name (e.g. `bash`, `read_file`, `gh`). The "friend" style already shows the real tool name — we should match that behaviour when there's only one entry.

## Expected behaviour

- **Single tool call in a turn** → show the actual tool name (e.g. `bash`, `write_file`, `gh pr review`)
- **Multiple tool calls** → current collapsing/grouping behaviour is fine

## Notes

Requested by @jmelloy via Discord. Likely lives in the Discord notifier's turn-formatting / tool-call summarisation logic.

### PR DESCRIPTION
**Pull request by jmelloy**
## Summary
- Turns with exactly one tool call now pass through their raw line (e.g. `▶ bash: pytest`) instead of collapsing into a generic `bash × 1` turn summary, matching the frontend's `LogList.vue` grouping which already skips grouping when `toolCallCount <= 1`.
- Turns with multiple tool calls keep the existing grouped `▶ Turn N: edit × 2, bash × 1` behavior unchanged.

## Test plan
- [x] Updated/added unit tests in `backend/tests/test_discord_notifier.py` covering single tool_use+tool_result, lone tool_use with no result, and zero-tool_use groups
- [x] `pytest tests/test_discord_notifier.py -k format_stream` passes
- [x] `ruff check` passes

Closes #886

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/discord_notifier.py [hunk 1]
```diff
@@ -1046,6 +1049,13 @@ def flush_group() -> None:
         nonlocal turn_number, group
         if not group:
             return
+        tool_use_count = sum(
```
Comment (jmelloy) [inline backend/discord_notifier.py]: The `tool_use_count` generator iterates `group` once per flush. For very large groups this is fine, but note that the subsequent `for entry in group` loop in the multi-tool path also iterates again. If `group` could ever be large, a single pass building `tool_use_count` and `counts` simultaneously would be more efficient. Not a bug, just worth noting.

File: backend/discord_notifier.py [hunk 2]
```diff
@@ -1046,6 +1049,13 @@ def flush_group() -> None:
...
+            1 for entry in group if (entry.detail or {}).get("toolType") == "tool_use"
+        )
+        if tool_use_count <= 1:
```
Comment (jmelloy) [inline backend/discord_notifier.py]: When `tool_use_count <= 1` the turn counter is NOT incremented, so turn numbering skips these groups entirely. This matches frontend behavior but means a sequence of [single-tool group, multi-tool group, multi-tool group] would produce Turn 1 and Turn 2 with no Turn 0 — which could confuse users correlating Discord threads with logs. Consider adding a brief inline comment clarifying this is intentional.

File: backend/tests/test_discord_notifier.py [hunk 3]
```diff
@@ -1395,16 +1395,59 @@ def test_format_stream_entries_passes_through_non_tool_lines():
...
+    ]
+
+    content, turn_count = discord_notifier._format_stream_entries(entries, 0)
+
+    assert content == "▶ bash: pytest -k test_foo\n  → 3 passed"
+    assert turn_count == 0
+
+
+def test_format_stream_entries_single_tool_call_with_no_result():
+    """A lone tool_use with no trailing tool_result still passes through raw."""
```
Comment (jmelloy) [inline backend/tests/test_discord_notifier.py]: This test name says 'with no result' but the scenario (a `tool_use` with no paired `tool_result`) is an unusual/malformed stream state. It's worth adding a comment noting whether this is an expected production scenario or a defensive edge case, so future maintainers understand the intent.

**Review by jmelloy — COMMENTED** [review]
- The core logic change is small and correct: groups with `tool_use_count <= 1` now pass through raw lines instead of generating a generic summary, matching the frontend's `LogList.vue` behavior.
- The turn counter is intentionally not incremented for single-tool-call groups, which is consistent with the frontend grouping but means turn numbers may have gaps if batches mix single and multi-tool groups — this should be documented or verified as intentional.
- The existing `test_format_stream_entries_numbers_turns_across_flushes` test was correctly updated to use multi-tool batches so it still exercises the numbering logic; the fix didn't silently break the test by changing semantics.
- The `tool_use_count` check uses `(entry.detail or {}).get('toolType') == 'tool_use'` which is consistent with how the rest of the function accesses detail, though a missing or malformed `detail` field will silently count as 0 rather than raising — acceptable defensive behavior here.
- New tests are well-scoped and cover the three meaningful edge cases (single use+result, lone use, zero use), giving good confidence in the boundary conditions.
